tree/tree.js

```diff
@@ -0,0 +1,43 @@
+
+class Tree {
+    constructor (value) {
+        this.value = value;
+        this.children = [];
+    }
+
+    addChild(value) {
+        const child = new Tree(value);
+        this.children.push(child);
+        return child;
+    }
+
+    contains(value) {
+        if (this.value === value) {
+            return true;
+        }
+        const len = this.children.length;
+        for (let i=0; i<len; i++) {
```

Comment:
can you try using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find . This can make your function shorter and more readable.
```javascript
this.childeren.find(child => child.contains(value));
```

Reply:
apparently, this should replace your 5 line of code.
```javascript
return this.childeren.find(child => child.contains(value));
```